cache.go

```diff
@@ -102,8 +115,19 @@ func (c *CacheConfig) validate() error {
 		return nil
 	}
 
-	if (c.Dir == nil && c.GoBuild == nil) || (c.Dir != nil && c.GoBuild != nil) {
-		return fmt.Errorf("invalid cache config: one of (and only one of) dir or gobuild must be set")
+	var count int
+	if c.Dir != nil {
+		count++
+	}
+	if c.GoBuild != nil {
+		count++
+	}
+	if c.Bazel != nil {
+		count++
+	}
+
+	if count != 1 {
+		return fmt.Errorf("invalid cache config: exactly one of (dir, gobuild, bazel) must be set")
```

Comment:
question: this message sounds like you can only use one cache for spec but I think that is not the case, right? It is really referring that on entry is specific to one type of cache? if so, could we add some sort not in the docs and maybe example of multiple cache usage?

Reply:
Note this is for a single cache config instance.
The higher level validation wraps this error.

Reply:
Updated docs to make this more clear.